docs/architecture/3rd party technical review/001-modi-poc.md

```diff
@@ -0,0 +1,114 @@
+ - `vp`
+ - `vs`
+ - `cd`
+ - `res`
+ - `f_pdf`, `f_qt`, `f_realp`, `f_wma`, `f_dir`, `f_fla`, `f_java`, `f_gears`, `f_ag`
+ - `refr`
+
+
+With the current architecture, it is worth mentionning that [IP adressess will be send](https://discourse.snowplowanalytics.com/t/running-snowplow-in-minimal-mode-for-gdpr/2391) to the collector. While the data will not be processed, the architecture need to be improved past the proof of concept to not leak those information.  
+
+### Recommended mitigations
+
+* Reduce the number of fields being sent to only ones that will be used.
+* Switch to a server side architecture enabling IP addresses of readers to not be shared 
+
+
+## Security engineering review
+
+### Web
+
+ Adding a third-party script with access to the main window involve several risks:
+  * Ability for the third-party script to modify content of the page 
+  * Ability for the third-party script to access cookies and local storage
+  * Ability for the third-party script to redirect user to a different page and domain
+  * Ability for the third-party script to fingerprint users
+  * Ability for the third-party script to send/retrieve data over the network
+
+Unfortunately our current security control are not enough to prevent most of those behaviours:
+ * Our CSP is allowing `unsafe-eval` and `unsafe-inline`
+ * We are not loading each third-party scripts in an iframe with reduced capabilities.
+ * We are not using [subresource intregrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) to ensure content of script is not changing. 
+
+Additionaly loading a third-party script from a storage not controlled by us, allow the content of the script to be modified from its original purpose for malicious activities.
+
+ #### Recommended mitigations
+
+Should the events being sent client-side:
+ * Audit the current version scrit ensuring it only does what is expected and creates a hash of it.
+ * Add [subresource intregrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) so the script could not be modified on the CDN maliciously
+
+
+### AMP
+
+TODO
+
+## Perfomance engineering review
+
+Adding another tracker sending regurlary will have a negative impact on performance. 
+However with the current scope no further analysis have been done.
+
+
+#### Recommended mitigations
+
+Outside of the proof of concept there are 2 options than can be investigated to ensure our performances
+are less degraded:
+ * Use [idle-until-urgent pattern](https://philipwalton.com/articles/idle-until-urgent/) using a library like [idlize](https://github.com/GoogleChromeLabs/idlize)
+ * Do not send the events client-side but server-side. This approach have several benefits:
+    - Do not require sending custom events in both apps and web platforms
+    - Ensure better privacy by design by not sharing any PII information such as IP addresses. 
+
```

Comment:
@jfsoul @nicl @gtrufitt I have shared my analysis for this third-party showing the kind of details expected in such a technical review. As the scope of this proof-of-concept is limited I have not really made a performance analysis but I would expect it to be the case should it not be one.

Reply:
Thanks this is very useful for us and other teams as a reference example.